src/components/Dashboard/FilingHistoryList.vue

```diff
@@ -1092,6 +1109,47 @@ export default {
 
       return (this.disableChanges || item.isCorrection || item.isFutureEffectiveIa || item.isColinFiling ||
         disableThisIaCorrection)
+    },
+
+    /**
+     * Fetches a filing by its id.
+     * @param businessId The business identifier
+     * @param filingId The filing identifier
+     * @returns a promise to return the filing of the specified type
+     */
+    fetchFilingById (businessId: string, filingId: number): Promise<any> {
```

Comment:
I suggest moving this to legal-api-mixin.ts (best practice, and so others can use this).

Reply:
I'm totally good with that, except this project doesn't have a legal-api mixin :)
There is a filing Mixin but it doesn't handle any requests, more just some functionality. Would you like me to create something similar here to what we use in the Create and edit ui?

Reply:
Yes, please create new mixin, but only for your one or 2 methods (no need to refactor all other code atm).
Baby steps towards a better code architecture! :)

Reply:
Absolutely :) No worries there lol

Reply:
Updated in next commit